http4s-client/src/main/scala/com/qvantel/jsonapi/client/http4s/Http4sClient.scala

```diff
@@ -0,0 +1,86 @@
+package com.qvantel.jsonapi.client.http4s
+
+import cats.Applicative
+import cats.effect._
+import com.netaporter.uri.config.UriConfig
+import com.netaporter.uri.{Uri => CoreUri}
+import com.netaporter.uri.dsl._
+import org.http4s.client.Client
+import org.http4s.dsl.io._
+
+import com.qvantel.jsonapi._
+import com.qvantel.jsonapi.client.http4s.JsonApiInstances._
+
+
+trait Http4sClient {
+  def apply[A](implicit jac: JsonApiClient[A]) = implicitly[JsonApiClient[A]]
+
+  private[this] def mkIncludeString(include: Set[String]): Option[String] = {
+    if (include.isEmpty) {
+      None
+    } else {
+      Some(include.mkString(","))
+    }
+  }
+
+  implicit def instance[A](implicit rt: ResourceType[A],
+                           reader: JsonApiReader[A],
+                           identifiable: Identifiable[A],
+                           endpoint: ApiEndpoint,
+                           client: Client[IO]): JsonApiClient[A] = new JsonApiClient[A] {
+
+    implicit val uConfig: UriConfig = uriConfig
+
+    override def one(id: String, include: Set[String]): IO[Option[A]] = {
+      for {
+        baseUri <- endpoint.uri
+        response <- pathOne(baseUri / rt.resourceType / id, include)
+      } yield response
+    }
+
+    override def many(ids: Set[String], include: Set[String]): IO[List[A]] = {
+      val requests = ids.map { id =>
+        one(id, include).flatMap {
+          case Some(entity) => IO.pure(entity)
+          case None         => IO.raiseError(ApiError.NoEntityForId(id, rt.resourceType))
+        }
+      }.toList
+
+      import cats.instances.list._
+
+      Applicative[IO].sequence(requests)
+    }
+
+    override def filter(filter: String, include: Set[String]): IO[List[A]] = {
+      implicit val _include: Include = Include(include)
+
+      for {
+        baseUri <- endpoint.uri
+        uri <- IO.fromEither(org.http4s.Uri.fromString((baseUri / rt.resourceType ? ("filter" -> filter) ? ("include" -> mkIncludeString(include))).toString))
```

Comment:
Do you really need `Uri.fromString` here?

Reply:
The [URI we use in other](https://github.com/lemonlabsuk/scala-uri) places is different from the one used by http4s. But I'll see if I can build it from the parts instead of going through strings

Reply:
Forgot it's the netaporter `Uri`, never mind.